README.md

```diff
@@ -8,3 +8,45 @@ SPDX-License-Identifier: Apache-2.0
 
 This repository contains the `.proto` files used by the Astarte message hub as well as by any
 client that wishes to connect itself to the message hub.
+
+## Code generation
+
+This project contains a Makefile that allows you to generate code for all supported languages.
+Currently, the language supported are:
+- Rust
+- Python.
+
+### Build 
+
+Build everything and generate the code for the various languages.
+```shell
+make
+```
+
+### Install 
+
+Install all language code files generated by the `all` rule in the repo folder.
+```shell
+make install
+```
+
+### Build language
+
+Build and generate code for specific language.
+```shell
+make rust
+```
```

Comment:
I would move this block below the `Build` one.

Reply:
I would like to keep this `build language step` separate from main the `build and install` because are extra commands.